File: main/src/main/java/de/blinkt/openvpn/core/CredentialsPopup.java
/*
 * Copyright (c) 2012-2019 Arne Schwabe
 * Distributed under the GNU GPL v2 with additional terms. For full terms see the file doc/LICENSE.txt
 */

package de.blinkt.openvpn.core;

import android.app.*;
import android.content.*;
import android.os.Bundle;
import android.os.IBinder;
import android.os.RemoteException;
import android.text.InputType;
import android.widget.EditText;
import de.blinkt.openvpn.R;

public class CredentialsPopup extends Activity {
    public static final String EXTRA_CHALLENGE_TXT = "de.blinkt.openvpn.core.CR_TEXT_CHALLENGE";




    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Get the alarm ID from the intent extra data
        Intent intent = getIntent();
        Bundle extras = intent.getExtras();
        if (extras == null) {
            finish();
            return;
        }
        String challenge = extras.getString(EXTRA_CHALLENGE_TXT, "(empty challenge text)");

        showPwDialog(challenge);
    }




    private void showPwDialog(String challenge) {
        DialogFragment frag = PasswordDialogFragment.newInstance(challenge);
        FragmentTransaction ft = getFragmentManager().beginTransaction();
        frag.show(ft, "dialog");
    }
}

File: main/src/ui/java/de/blinkt/openvpn/activities/CredentialsPopup.kt
/*
 * Copyright (c) 2012-2019 Arne Schwabe
 * Distributed under the GNU GPL v2 with additional terms. For full terms see the file doc/LICENSE.txt
 */
package de.blinkt.openvpn.activities

import android.content.Intent
import android.net.VpnService
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import de.blinkt.openvpn.core.OpenVPNService
import de.blinkt.openvpn.core.PasswordDialogFragment.Companion.newInstance

class CredentialsPopup : AppCompatActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // Get the alarm ID from the intent extra data
        val intent = intent
        showPwDialog(intent)
    }

    private fun showPwDialog(intent: Intent) {
        val frag: DialogFragment? = newInstance(intent, true)
        if (frag == null) {
            finish()
            return
        }
        frag.show(supportFragmentManager, "dialog")
    }
}